crates/types/src/config/cli_option_overrides.rs

```diff
@@ -97,24 +109,21 @@ pub struct CommonOptionCliOverride {
     #[clap(long, global = true)]
     pub advertise_address: Option<AdvertisedAddress>,
 
-    /// # Partitions
+    /// Default Number Of Partitions
+    ///
+    /// Number of partitions that will be provisioned during initial cluster provisioning.
+    /// partitions are the logical shards used to process messages. Default is 24.
     ///
-    /// Number of partitions that will be provisioned during cluster bootstrap,
-    /// partitions used to process messages.
+    /// Cannot be higher than `65535` (You should almost never need as many partitions anyway)
     ///
-    /// NOTE: This config entry only impacts the initial number of partitions, the
+    /// NOTE 1: This config entry only impacts the initial number of partitions, the
     /// value of this entry is ignored for bootstrapped nodes/clusters.
     ///
-    /// Cannot be higher than `4611686018427387903` (You should almost never need as many partitions anyway)
-    #[clap(long, global = true)]
-    pub bootstrap_num_partitions: Option<NonZeroU64>,
-
-    /// # Automatically provision number of configured partitions
+    /// NOTE 2: This will be renamed to `default-num-partitions` by default as of v1.3+
     ///
-    /// If this option is set to `false`, then one needs to manually write a partition table to
-    /// the metadata store. Without a partition table, the cluster will not start.
-    #[clap(long, global = true)]
-    pub auto_provision_partitions: Option<bool>,
+    /// Default: 24
+    #[clap(long, global = true, alias = "bootstrap-num-partitions")]
+    pub default_num_partitions: Option<NonZeroU64>,
```

Comment:
This breaks in a weird way when set via an env variable:

```
❯ RESTATE_BOOTSTRAP_NUM_PARTITIONS=1 restate-server
[RT0002] configuration loading error: duplicate field `default-num-partitions`
```

Reply:
Update: rebuilt from the PR branch and tested again, this no longer happens. Both of these work as expected:

```
RESTATE_DEFAULT_NUM_PARTITIONS=1 restate-server
RESTATE_BOOTSTRAP_NUM_PARTITIONS=1 restate-server
```